Example/AsyncReactorExample/Features/Repository/Search/RepositorySearchView.swift

```diff
@@ -37,24 +38,36 @@ struct RepositorySearchView: View {
                 
                 if !reactor.repositories.isEmpty {
                     RepositoryList(
-                        repositories: reactor.repositories.filter { !hidePrivate || $0.isVisible }
+                        repositories: reactor.repositories
+                            .filter {
+                                !hidePrivate || $0.isVisible
+                            }
+                            .sorted(by: {
+                                switch sortOption {
+                                case .watchers:
+                                    return $0.watchersCount < $1.watchersCount
+                                case .forks:
+                                    return $0.forks < $1.forks
+                                }
+                            })
                     )
                 }
             }
             .navigationTitle("Repositories")
             .refreshable {
                 reactor.send(.load)
             }
-            
             .searchable(text: $query)
-            .onChange(of: query) { query in
-                queryPublisher.send(query)
-            }
-            .onReceive(
-                queryPublisher
-                    .debounce(for: .seconds(1), scheduler: DispatchQueue.main)
-            ) { query in
-                reactor.send(.load)
+            .toolbar {
+                ToolbarItem {
+                    Menu("Sort By") {
+                        Picker("Sort By", selection: $sortOption) {
+                            ForEach(SortOptions.allCases, id: \.self) {
```

Comment:
Its cleaner to make the SortOptions conform to Identifiable. You can use the rawValue as id. Then you don't need to provide the id here